### PR TITLE
Ch signup documentation

### PR DESCRIPTION
**Pull request by Hurntre**
#### Description
the documentation was updated to include the signup route. then password hash and token generation helpers was added
#### Type of change

- [ ]
- [ ]

#### How Has This Been Tested?
updated the signup validation tests to include the token key
- [ ] Unit testing

#### PT-ID

[](url)

#### Screenshots
image

#### Questions:

### CODE REVIEW

File: server/helpers/auth.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import bcrypt from 'bcryptjs';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const authHelper = {
+  hashPassword: password => {
+    return bcrypt.hashSync(password, 10);
+  },
+  comparePassword: (hashPassword, password) => {
```
Comment (vic3king) [inline server/helpers/auth.js]: NitPick: `hashPassword` should be `hashedPassword`

File: tests/integration/signupValidation.test.js [hunk 2]
```diff
@@ -34,6 +34,7 @@ describe('User Creation', () => {
           'phoneNumber',
           'password',
           '__v',
+          'token',
```
Comment (vic3king) [inline tests/integration/signupValidation.test.js]: Kindly write an assertion to ensure that when a user is created a new token is returned.

File: server/db/models/user.js [hunk 3]
```diff
@@ -1,5 +1,17 @@
 import mongoose from 'mongoose';
 import userSchema from '../schemas/userSchema';
+import authHelper from '../../helpers/auth';
+import tokenGenerator from '../../helpers/userToken';
+
+// eslint-disable-next-line func-names
```
Comment (vic3king) [inline server/db/models/user.js]: why are we disabling this rule?

File: documentation/swagger.json [hunk 4]
```diff
@@ -2,35 +2,105 @@
...
+            "name": "email",
+            "in": "formData",
+            "description": "email of the user",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "phoneNumber",
+            "in": "formData",
+            "description": "phone name",
```
Comment (vic3king) [inline documentation/swagger.json]: *Phone number

File: server/db/models/user.js [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+userSchema.pre('validate', async function() {
+  if (this.isNew) {
+    const hashedPassword = await authHelper.hashPassword(this.password);
```
[the remark this answers is not in the document]
Comment (Hurntre) [reply server/db/models/user.js]: thanks

File: package.json [hunk 6]
```diff
@@ -54,6 +54,7 @@
     "@babel/register": "^7.4.0",
     "@hapi/joi": "^16.1.7",
     "babel-eslint": "^10.0.1",
+    "bcrypt": "^3.0.7",
     "bcryptjs": "^2.4.3",
```
[the remark this answers is not in the document]
Comment (Hurntre) [reply package.json]: done

File: server/helpers/auth.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import bcrypt from 'bcryptjs';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const authHelper = {
+  hashPassword: password => {
```
[the remark this answers is not in the document]
Comment (Hurntre) [reply server/helpers/auth.js]: thanks

File: server/helpers/auth.js (hunk 1 above)
Comment (Hurntre) [reply server/helpers/auth.js]: done

File: tests/integration/signupValidation.test.js [hunk 8]
```diff
@@ -39,6 +39,27 @@ describe('User Creation', () => {
...
+        email: 'emailqwet@yahoo.com',
+        phoneNumber: '08099999999',
+        password: 'pAsSwOrD',
+        confirmPassword: 'pAsSwOrD',
+      })
+      .end((err, res) => {
+        const { success, error, token } = res.body;
+        expect(res).to.have.status(200);
+        expect(success).to.equal(true);
+        expect(token).to.not.equal(undefined && null);
```
Comment (vic3king) [inline tests/integration/signupValidation.test.js]: NIce!

**Pull request merged**